src/helper/class-ee-site.php

```diff
@@ -1177,6 +1180,32 @@ public function populate_site_info( $site_name, $in_array = true ) {
 		$this->site_data = EE\Site\Utils\get_site_info( [ $site_name ], false, false, $in_array );
 	}
 
+	/**
+	 * Allow custom SSL for site.
+	 *
+	 * @param $ssl_key string path of the custom ssl key.
+	 * @param $ssl_crt string path of the custom ssl cert.
+	 *
+	 * @throws EE\ExitException
+	 */
+	public function custom_site_ssl( $ssl_key, $ssl_crt ) {
+
+		if ( empty( $ssl_key ) || empty( $ssl_crt ) ) {
+			EE::error( 'Pass --ssl-key and --ssl-crt for custom SSL' );
```

Comment:
Throw an exception instead of using error. Throwing exception will show the missing info, as well as trigger cleanup of site. Error will halt site creation, leaving extra files, containers and networks lying around.